app/questionnaire/questionnaire_manager.py

```diff
@@ -31,13 +31,13 @@ def __init__(self, schema, json=None):
         self._schema = schema
         self.state = None
 
-    def validate(self, location, post_data):
+    def validate(self, location, post_data, skip_mandatory_validation=False):
```

Comment:
Not sure what else we can do till WTF time as at the top level its just an array of messages with no validation type associated, so I can't tell if they are mandatory failures